packages/worker/src/runtime.ts

```diff
@@ -283,7 +282,7 @@ export class Runtime {
    * @hidden
    */
   public async closeNativeClient(client: native.Client): Promise<void> {
-    this.registeredClients.delete(client);
+    this.backRefs.delete(client);
```

Comment:
Not sure if there is actually a risk here, but removing `client` from `backrefs` before making the call to `clientClose` means that there is no protection against tokio runtime destruction while in the `clientClose()` function (line 326). Compare this to line 317 and 463.

Reply:
There's no risk because it's sync JS and therefore atomic, but I'll change this for consistency